bot-components/GitHub_queries.ml

```diff
@@ -267,6 +267,82 @@ let get_pull_request_refs ~bot_info ~owner ~repo ~number =
           f "Query pull_request_info failed with %s" err)
   >|= Result.bind ~f:(pull_request_info_of_resp ~owner ~repo ~number)
 
+let merge_pull_request_info_of_resp ~owner ~repo ~number resp :
+    (GitHub_subscriptions.merge_pull_request_info, string) Result.t =
+  match resp#repository with
+  | None ->
+      Error (f "Unknown repository %s/%s." owner repo)
+  | Some repository -> (
+    match repository#pullRequest with
+    | None ->
+        Error (f "Unknown pull request %s/%s#%d." owner repo number)
+    | Some pull_request -> (
+      match pull_request#assignees#nodes with
+      | None ->
+          Error "No assignees found."
+      | Some assignees -> (
+        match pull_request#milestone with
+        | None ->
+            Error "No milestone found."
+        | Some milestone -> (
+          match pull_request#labels with
+          | None ->
+              Error "No labels found."
+          | Some labels -> (
+            match pull_request#reviews with
+            | None ->
+                Error "No reviews found."
+            | Some reviews ->
```

Comment:
You could make this code more compact by matching over several fields at once, e.g.:
```suggestion
      match pull_request#assignees#nodes, pull_request#milestone, pull_request#labels, pull_request#reviews with
      | None, _, _, _ ->
          Error "No assignees found."
      | _, None, _, _ ->
          Error "No milestone found."
      | _, _, None, _ ->
          Error "No labels found."
      | _, _, _, None ->
          Error "No reviews found."
      | Some assignees, Some milestone, Some labels, Some reviews ->
```